src/content/layout/builtin_layout.cc

```diff
@@ -51,6 +51,74 @@
+    chain["/Photos/Directories"] = this->content->addContainerTree(std::vector<std::shared_ptr<CdsObject>> { container["Photos"], container["Photos/Directories"] });
+
+    container["Audio"] = std::make_shared<CdsContainer>("Audio");
+    container["All Audio"] = std::make_shared<CdsContainer>("All Audio");
+    container["All Songs"] = std::make_shared<CdsContainer>("All Songs");
+    chain["/Audio/All Audio"] = this->content->addContainerTree(std::vector<std::shared_ptr<CdsObject>> { container["Audio"], container["All Audio"] });
+    container["All - full name"] = std::make_shared<CdsContainer>("All - full name");
+    chain["/Audio/All - full name"] = this->content->addContainerTree(std::vector<std::shared_ptr<CdsObject>> { container["Audio"], container["All - full name"] });
+    container["Albums"] = std::make_shared<CdsContainer>("Albums");
+    chain["/Audio/Albums"] = this->content->addContainerTree(std::vector<std::shared_ptr<CdsObject>> { container["Audio"], container["Albums"] });
+    container["Artists"] = std::make_shared<CdsContainer>("Artists");
+    chain["/Audio/Artists"] = this->content->addContainerTree(std::vector<std::shared_ptr<CdsObject>> { container["Audio"], container["Artists"] });
+    container["Genres"] = std::make_shared<CdsContainer>("Genres");
+    chain["/Audio/Genres"] = this->content->addContainerTree(std::vector<std::shared_ptr<CdsObject>> { container["Audio"], container["Genres"] });
+    container["Composers"] = std::make_shared<CdsContainer>("Composers");
+    chain["/Audio/Composers"] = this->content->addContainerTree(std::vector<std::shared_ptr<CdsObject>> { container["Audio"], container["Composers"] });
+    container["Year"] = std::make_shared<CdsContainer>("Year");
+    chain["/Audio/Year"] = this->content->addContainerTree(std::vector<std::shared_ptr<CdsObject>> { container["Audio"], container["Year"] });
+    container["Audio/Directories"] = std::make_shared<CdsContainer>("Directories");
+    chain["/Audio/Directories"] = this->content->addContainerTree(std::vector<std::shared_ptr<CdsObject>> { container["Audio"], container["Audio/Directories"] });
+
+#ifdef ONLINE_SERVICES
+    container["Online Services"] = std::make_shared<CdsContainer>("Online Services");
+#ifdef SOPCAST
+    container["SopCast"] = std::make_shared<CdsContainer>("SopCast");
+    container["SopCast/All Channels"] = std::make_shared<CdsContainer>("All Channels");
+    container["SopCast/Groups"] = std::make_shared<CdsContainer>("Groups");
+    chain["/Online Services/SopCast/All Channels"] = this->content->addContainerTree(std::vector<std::shared_ptr<CdsObject>> { container["Online Services"], container["SopCast"], container["SopCast/All Channels"] });
+    chain["/Online Services/SopCast/Groups"] = this->content->addContainerTree(std::vector<std::shared_ptr<CdsObject>> { container["Online Services"], container["SopCast"], container["SopCast/Groups"] });
+#endif
+
+#ifdef ATRAILERS
```

Comment:
Would it be possible to convert these to if constexpr?

Reply:
They are spread all over the codebase.

Reply:
Different PR then.